content/en/infrastructure/containers/configuration.md

```diff
@@ -266,13 +266,129 @@ Follow these steps to collect the custom resources that these CRDs define:
 
-   <div class="alert alert-info">You can select a maximum of 50 fields for each resource.</div>
+#### Indexing complex types
+
+For arrays of objects, two indexing strategies are available:
+
+-   **Field**: Groups object array nested fields under each unique value of the selected field.
+-   **No field**: Flattens object array nested fields into arrays, merging all field values.
+
+##### Example: Filtering on DatadogPodAutoscaler custom resources
+
+Consider these two custom resources:
+
+{{< tabs >}}
+{{% tab "CR1" %}}
+
+```yaml
+status:
+    conditions:
+        - type: HorizontalAbleToScale
+          status: 'True'
+        - type: VerticalAbleToApply
+          status: 'False'
+```
+
+{{% /tab %}}
+{{% tab "CR2" %}}
+
+```yaml
+status:
+    conditions:
+        - type: VerticalAbleToApply
+          status: 'True'
+        - type: HorizontalAbleToScale
+          status: 'False'
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+You have the filtering possibilities on `status.conditions` based on the two indexing strategies:
+
+{{< tabs >}}
+{{% tab "Grouping by no field" %}}
+
+**Indexed fields for CR1:**
+
+```yaml
+status:
+    conditions:
+        type: [HorizontalAbleToScale, VerticalAbleToApply]
+        status: ['True', 'False']
+```
+
+**Indexed fields for CR2:**
+
+```yaml
+status:
+    conditions:
+        type: [VerticalAbleToApply, HorizontalAbleToScale]
+        status: ['True', 'False']
+```
+
+**Example queries:**
+
+**Query 1:**
+
+```text
+field#status.conditions.status:"False"
+```
+
+**Result:** Returns CR1 and CR2. Both CRs have at least one object with `status:"False"`
+
+**Query 2:**
+
+```text
+field#status.conditions.status:"False" AND field#status.conditions.type:VerticalAbleToApply
+```
+
+**Result:** Returns CR1 and CR2. Each filter is evaluated separately. At least one `status.condition` object in each custom resource matches each filter, even if it's not the same object that matches both filters.
+
+{{% /tab %}}
+{{% tab "Grouping by type" %}}
+
+**Indexed fields for CR1:**
+
+```yaml
+status:
+    conditions:
+        - HorizontalAbleToScale:
+              status: 'True'
+        - VerticalAbleToApply:
+              status: 'False'
+```
+
+**Indexed fields for CR2:**
+
+```yaml
+status:
+    conditions:
+        - VerticalAbleToApply:
+              status: 'True'
+        - HorizontalAbleToScale:
+              status: 'False'
+```
+
+**Example query:**
+
+```text
+field#status.conditions.HorizontalAbleToScale.status:"False"
+```
+
+**Result:** Returns only CR2. The query filters for CRs where `status.condition` objects of type `HorizontalAbleToScale` have `status:"False"`. Both the `type` and `status` filters must match the same object.
+
+{{% /tab %}}
+{{< /tabs >}}
+
+<div class="alert alert-info">You can select up to 50 fields per resource. You can use the preview to validate your indexing choices.</div>
+
+Select **Enable Indexing** to save.
```

Comment:
Should this be a separate step, i.e. Step 5, following this one?

<img width="762" height="71" alt="image" src="https://github.com/user-attachments/assets/819a2b0b-0d67-4b81-8207-c144f1c453d1" />